scm-core/src/main/java/sonia/scm/update/RepositoryUpdateIterator.java

```diff
@@ -39,4 +41,35 @@
    * @since 2.13.0
    */
   void forEachRepository(Consumer<String> repositoryIdConsumer);
+
+  /**
+   * Equivalent to {@link #forEachRepository(Consumer)} with the difference, that you can throw exceptions in the given
+   * update code, that will then be wrapped in a {@link UpdateException}.
+   *
+   * @since 2.14.0
+   */
+  default void forEachRepositoryWithException(UpdaterWithException updater) {
```

Comment:
I don't like this methode name. It sounds like it throws an exception no matter what.